app/src/main/java/nl/avans/ti/MainActivity.java

```diff
@@ -1,14 +1,58 @@
 package nl.avans.ti;
 
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
 import androidx.appcompat.app.AppCompatActivity;
 
-import android.os.Bundle;
+import nl.avans.ti.Medal.MedalActivity;
+
+public class MainActivity extends AppCompatActivity
+{
+
 
-public class MainActivity extends AppCompatActivity {
+    EditText editText;
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    protected void onCreate(Bundle savedInstanceState)
+    {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        editText = findViewById(R.id.editTextNumberSigned);
+        Button button = findViewById(R.id.button);
+
+
+    }
+
+
+    public void onButtonPressed(View view)
+    {
+        String enteredCode = editText.getText().toString();
+
+        if (enteredCode.length() < 4)
+        {
+            Toast.makeText(view.getContext(), "de code moet 4 cijfers lang zijn", Toast.LENGTH_LONG).show();
```

Comment:
De code moet 6 cijfers lang zijn
```suggestion
            Toast.makeText(view.getContext(), "de code moet 6 cijfers lang zijn", Toast.LENGTH_LONG).show();
```